bond/bond_test.go

```diff
@@ -38,110 +40,107 @@ const (
 			"cniVersion": "%s",
 			"mode": "%s",
 			"failOverMac": 1,
-			"linksInContainer": true,
+			"linksInContainer": %s,
 			"miimon": "100",
-			"mtu": 1400,
+			"mtu": %s,
 			"links": [
 				{"name": "net1"},
 				{"name": "net2"}
 			]
 		}`
 	ActiveBackupMode = "active-backup"
 	BalanceTlbMode   = "balance-tlb"
+	DefaultMTU       = "1400"
 )
 
 var Slaves = []string{Slave1, Slave2}
 
 var _ = Describe("bond plugin", func() {
 	var podNS ns.NetNS
+	var linkAttrs []netlink.LinkAttrs
+	var hostNS ns.NetNS
+	var linksInContainer bool
 
-	BeforeEach(func() {
+	JustBeforeEach(func() {
 		var err error
 		podNS, err = testutils.NewNS()
 		Expect(err).NotTo(HaveOccurred())
-
-		for _, ifName := range Slaves {
-			err = podNS.Do(func(ns.NetNS) error {
-				defer GinkgoRecover()
-				err = netlink.LinkAdd(&netlink.Dummy{
-					LinkAttrs: netlink.LinkAttrs{
-						Name: ifName,
-					},
-				})
-				Expect(err).NotTo(HaveOccurred())
-				_, err := netlink.LinkByName(ifName)
-				Expect(err).NotTo(HaveOccurred())
-				return nil
-			})
+		if linksInContainer {
+			addLinksInNS(podNS, linkAttrs)
+		} else {
+			hostNS, err = ns.GetCurrentNS()
```

Comment:
I think the main reason to not use the host namespace was to not have issues when something fails to clean up.

What happens if the dummies interfaces cannot be deleted in the host namespace? Will subsequent tests (with linkInContainer = false) fail because the add command will complain that the interfaces already exist?

Reply:
i agree with you, didn't think about it. I added another namespace (initNS) instead of using the host.